### ISSUE TITLE
Bug in the RelaX SQL that crashes Chrome and Firefox

### ISSUE THREAD

**Issue by Henriquemcc** [dbis-uibk/relax#116]
On the RelaX - relational algebra calculator, when I load the IMDB database and type the following SQL command:
```
SELECT movies.name, actors.first_name, roles.role
FROM actors, movies, roles
WHERE actors.id = roles.actor_id
AND movies.id = roles.movie_id
AND movies.rank > 6
```
 and them click on 'execute query', on both Firefox and Google Chrome, the RelaX web page stop responding.

**Comment by olberger** [comment]
AFAIU, the joins aren't detected well, and this leads to a cartesian product being applied, which results in OOM or some other kind of combinatory explosion.

I'd suggest to test with an INNER JOIN variant, to see if that works, instead.

Hth,

**Comment by r-prem** [comment]
As @olberger correctly stated, it is because of the cross joins. Included in the commit eda00d0 is a warning if the app thinks a OOM error might occur.
